### PR TITLE
Dynamic import syntax highlighter from source to save on bundle size

### PR DESCRIPTION
**Pull request by plesiecki**
https://github.com/allegro/turnilo/issues/845

### CODE REVIEW

**Review by adrianmroz-allegro — APPROVED** [review]
I need to keep reference to this PR so I will remember why we have this `exclude` clauses and where we need to modify them in the future

File: src/client/modals/source-modal/highlighter.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+export default class Highlighter extends React.Component<Props> {
+    render() {
+        const { children: source } = this.props;
+
+        return <SyntaxHighlighter className="source-modal__source" language="json" style={githubGist}>
+            {source}
+        </SyntaxHighlighter>;
+    }
+}
```
Comment (adrianmroz-allegro) [inline src/client/modals/source-modal/highlighter.tsx:35]:
```suggestion
export const Highlighter: SFC = ({ children: source }) =>
  <SyntaxHighlighter className="source-modal__source" language="json" style={githubGist}>
      {source}
  </SyntaxHighlighter>;
```

Comment (adrianmroz-allegro) [reply src/client/modals/source-modal/highlighter.tsx:35]: Of course, need to import SFC from React.

File: src/client/modals/source-modal/source-modal.tsx [hunk 2]
```diff
@@ -57,9 +55,11 @@ export class SourceModal extends React.Component<SourceModalProps, SourceModalSt
     >
       <div className="content">
         {header}
-        <SyntaxHighlighter className="source-modal__source" language="json" style={githubGist}>
-          {source}
-        </SyntaxHighlighter>
+        <React.Suspense fallback={Loader}>
```
Comment (adrianmroz-allegro) [inline src/client/modals/source-modal/source-modal.tsx:58]: Perfect case for lazy component!

File: src/client/modals/source-modal/highlighter.tsx (hunk 1 above)
Comment (plesiecki) [reply src/client/modals/source-modal/highlighter.tsx:35]: TS told me `SFC` is deprecated and I should use `FunctionComponent`.
Also it has to export default to work as lazy component.

File: src/client/modals/source-modal/source-modal.tsx (hunk 2 above)
Comment (plesiecki) [reply src/client/modals/source-modal/source-modal.tsx:58]: Yeah, it is! Wonder if `SourceModal` wouldn't be a better candidate though.

File: src/client/modals/source-modal/highlighter.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * Luckily untranspiled source code is avaiable within npm package. */
+import SyntaxHighlighter from "react-syntax-highlighter/src/light";
+import githubGist from "react-syntax-highlighter/src/styles/hljs/github-gist";
+
+SyntaxHighlighter.registerLanguage("json", jsonLanguage);
+
+interface Props {
+}
+
+const Highlighter: React.FunctionComponent<Props> = ({ children: source }) =>
```
[the remark this answers is not in the document]
Comment (plesiecki) [reply src/client/modals/source-modal/highlighter.tsx]: done

**Pull request merged**